lib/webpacker/dev_server.rb

```diff
@@ -0,0 +1,91 @@
+# Same convention as manifest/configuration.rb
+
+# Loads webpacker configuration from config/webpacker.yml
+
+require "webpacker/configuration"
+
+class Webpacker::DevServer < Webpacker::FileLoader
+  class << self
+    def dev_server?
+      env_val = env_value("WEBPACKER_DEV_SERVER")
+
+      # override dev_server setup WEBPACKER_DEV_SERVER=FALSE
+      return false if env_val == false
+
+      # If not specified, then check if values in the config file for the dev_server key
+      !dev_server_values.nil?
+    end
+
+    # read settings for dev_server
+    def hot?
+      return false unless dev_server?
+      env_val = env_value("WEBPACKER_HMR")
+      return env_val unless env_val.nil?
+      fetch(:hot)
```

Comment:
Let's start by renaming to `hot_reloading?` and then we can simplify the implementation:

```ruby
if enabled?
  case ENV["WEBPACKER_HMR"]
  when /true/i then true
  when /false/i then false
  else fetch(:hot_reloading)
  end
end
```